grails-app/controllers/org/pih/warehouse/api/StockMovementApiController.groovy

```diff
@@ -369,11 +370,11 @@ class StockMovementApiController {
 
         // Dates aren't bound properly using default JSON binding
         if (jsonObject.containsKey("dateShipped")) {
-            stockMovement.dateShipped = DateUtil.asDate(jsonObject.remove("dateShipped") as String)
+            stockMovement.dateShipped = dateParser.parseToDate(jsonObject.remove("dateShipped"))
```

Comment:
an example of the new DateParser in use. It preserves the existing behaviour while making the actual refactoring to java.time classes easier. Once we move away from Date, we can simply change this to `dateParser.parseToInstant(...)`. An easy refactor